WordPress/Classes/ViewRelated/Stats/Insights/ViewsVisitors/ViewsVisitorsLineChartCell.swift

```diff
@@ -54,14 +54,34 @@ struct StatsSegmentedControlData {
     }
 
     var differenceLabel: String {
-        let stringFormatValue = differencePercent != 0 ? "%@%@ (%@%%)" : "%@%@"
-        let stringFormat = NSLocalizedString(stringFormatValue, comment: "Difference label for Insights Overview stat, indicating change from previous period. Ex: +99.9K(5%)")
-        return String.localizedStringWithFormat(
-            stringFormat,
-            difference < 0 ? "" : "+",
-            difference.abbreviatedString(),
-            differencePercent.abbreviatedString()
-        )
+        // We want to show something like "+1.2K (5%)" if we have a percentage difference and "1.2K" if we don't.
+        // Because localized strings need to be strings literal, we cannot embed any conditional logic in the `localizedString...` call.
+        // We therefore need to generate different string literals base on the state.
+        let differenceSign = difference < 0 ? "" : "+"
```

Comment:
Even that conditional about `differenceSign` is making assumptions about the fact that those two variants would always be translated the same with just the added `+` in all languages — while depending on how different locales decide to display those values according to locale rules and practices, that might not be true (e.g. either they might put the `+` sign elsewhere like after instead of before in some locales, or they might have some other character for those cases, just like Arabic has a different Unicode character for the % sign, aka U+066A, or something else…)

So personally I'd even suggest to provide 4 different strings and not just 2, to cover all cases while allowing polyglots to still provide all variants matching their locale rules.

(See also: pbMoDN-3tH-p2#the-tips)

Reply:
Good point, we shouldn't be assuming particular symbols or their order in different languages.

Reply:
Good point. On it...